Source/SessionManager.swift

```diff
@@ -250,33 +247,45 @@ open class SessionManager {
     /// - parameter urlRequest: The URL request.
     ///
     /// - returns: The created `DataRequest`.
-    open func request(resource urlRequest: URLRequestConvertible) -> DataRequest {
-        let originalRequest = urlRequest.urlRequest
-        let originalTask = DataRequest.Requestable(urlRequest: originalRequest)
+    open func request(_ urlRequest: URLRequestConvertible) -> DataRequest {
+        do {
+            let originalRequest = try urlRequest.asURLRequest()
+            let originalTask = DataRequest.Requestable(urlRequest: originalRequest)
 
-        let task = originalTask.task(session: session, adapter: adapter, queue: queue)
-        let request = DataRequest(session: session, task: task, originalTask: originalTask)
+            let task = try originalTask.task(session: session, adapter: adapter, queue: queue)
+            let request = DataRequest(session: session, requestTask: .data(originalTask, task))
 
-        delegate[request.delegate.task] = request
+            delegate[task] = request
 
-        if startRequestsImmediately { request.resume() }
+            if startRequestsImmediately { request.resume() }
```

Comment:
Perhaps this could be put in a `defer` block before the `do` so it's not called both here and in `request(failedWith:)`?